### PR TITLE
add integration tests

### PR DESCRIPTION
**Pull request by sarge**
Clean up and reapply of the of integration tests.

### CODE REVIEW

**Comment by mholt** [comment]
Thanks Sarge! I'll get around to taking a close look as soon as I can catch up on the PKI stuff.

**Comment by sarge** [comment]
@mholt I think these commits are missing from master - https://github.com/sarge/caddy/commits/reload_certificate_files which is causing this to fail

File: azure-pipelines.yml [hunk 1]
```diff
@@ -130,6 +130,16 @@ jobs:
...
     displayName: Run tests
 
+  - script: |
+      set -e
+      cmd/caddy/caddy start
+      go test -v -count=1 ./caddytest/...
+      cmd/caddy/caddy stop
+    workingDirectory: '$(modulePath)'
+    continueOnError: false
+    condition: or(eq( variables['Agent.OS'], 'Linux' ),eq( variables['Agent.OS'], 'Darwin' ))
```
[the remark this answers is not in the document]
Comment (sarge) [reply azure-pipelines.yml]: Correct, good catch

**Comment by sarge** [comment]
Hmm why no CI tests?

**Comment by sarge** [comment]
Merged as part of separate PR

**Pull request closed without merging**